rdr_service/api/nph_participant_api_schemas/schema.py

```diff
@@ -355,13 +364,22 @@ class Meta:
         connection_class = ParticipantConnection
 
     participant = relay.ConnectionField(
-        ParticipantConnection, nph_id=String(required=False), sort_by=String(required=False), limit=Int(required=False),
-        off_set=Int(required=False),
+        ParticipantConnection,
+        nph_id=String(required=False),
+        sort_by=String(required=False),
+        limit=Int(required=False, default_value=DEFAULT_LIMIT),
+        off_set=Int(required=False, default_value=DEFAULT_OFFSET),
         **_build_filter_parameters(Participant)
     )
 
     @staticmethod
     def resolve_participant(root, info, nph_id=None, sort_by=None, limit=None, off_set=None, **filter_kwargs):
+        # Set the value of pagination 'limit' between 1 (min), 1000 (max) & 100 (default)
+        limit = min(max(limit, MIN_LIMIT), MAX_LIMIT)
+
+        # Set the value of pagination 'off_set' between 0 (min), 1000 (max) & 0 (default)
+        off_set = min(max(off_set, MIN_OFFSET), MAX_OFFSET)
```

Comment:
Is there a need for a max offset? Limiting the number of records that can go out in a response helps keep processing times shorter, but I'm not sure what limiting the offset would do. Offset determines how far down the list of participants they can go, rather than how much we're sending back in one response.

This does seem to keep requests from being able to get all the data. If we have about 5k participants, then the offset would mean any clients will only be able to see data for the first 1k.

Reply:
I see your point. I added that to prevent the user from sending any erroneously large offset value. I will remove the max offset.